src/workspacer.Shared/Layout/FloatingLayoutEngine.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace workspacer
+{
+    public class FloatingLayoutEngine : ILayoutEngine
+    {
+        // we need to account for the Bar height (which is by default 30 height), otherwise, windows will be "dragged down"
+        private readonly int _barHeight;
+
+        public FloatingLayoutEngine() : this(30) { }
+
+        public FloatingLayoutEngine(int barHeight)
+        {
+            _barHeight = barHeight;
+        }
+
+        public IEnumerable<IWindowLocation> CalcLayout(IEnumerable<IWindow> windows, int spaceWidth, int spaceHeight)
+        {
+            var list = new List<IWindowLocation>();
+            var numWindows = windows.Count();
+
+            if (numWindows == 0)
+                return list;
+
+            var windowList = windows.ToList();
+
+            for (var i = 0; i < numWindows; i++)
+            {
+                var window = windowList[i];
+
+                list.Add(new WindowLocation(
+                            window.Location.X-window.Offset.X,
+                            window.Location.Y-window.Offset.Y-_barHeight,
```

Comment:
I've encountered an issue with a secondary monitor to the right of the primary monitor, where a window will continually be shifted further to the right and thus hidden.

I think it's because the `WindowLocation`'s `X` and `Y` positions are being updated here:
https://github.com/rickbutton/workspacer/blob/0aaff4437222e471ea0ffd282b948a9dfb8b7ff5/src/workspacer.Shared/Workspace/Workspace.cs#L353-L354

I think this can be fixed by:
```suggestion
                            0, 0,
```

That also means `_barHeight` isn't necessary.

Reply:
I currently don't have multiple monitors, so it will take me a few days until I can test it.

The exact issue you've experienced happens out of the box if I don't use the `_barHeight` variable using a single monitor setup. That single variable was the reason why the floating layout worked in the single monitor setup.

Your suggestion will snap the window to the top left corner of the screen(?). I'll take a deeper look at the multi-monitor floating setup to make sure it works with more monitors.

Rather, to prevent shifting I think the right "equation" would be:

`window.Location.Y-window.Offset.Y- (<number of monitors> * _barHeight)`